src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IBranchOperation.Extensions.cs

```diff
@@ -0,0 +1,416 @@
+            Assert.Null(actual);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [WorkItem(28095, "https://github.com/dotnet/roslyn/issues/28095")]
+        [Fact]
+        public void GetCorrespondingLoop_BreakButCorrespondingToSwitch_ReturnsNull()
+        {
+            var (outer, actual) = GetOuterOperationAndCorrespondingInnerOperation<ForStatementSyntax, ILoopOperation, BreakStatementSyntax>(@"
+class C
+{
+    void F()
+    {
+        /*<bind>*/for (;;)
+        {
+            switch (1)
+            {
+                case 1:
+                /*<bind>*/break;/*</bind>*/
+            }
+        }/*</bind>*/
+        
+    }
+}", branch => branch.GetCorrespondingLoop());
+
+            Assert.NotNull(outer);
+            Assert.Null(actual);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [WorkItem(28095, "https://github.com/dotnet/roslyn/issues/28095")]
+        [Fact]
+        public void GetCorrespondingSwitch_BreakButCorrespondingToLoop_ReturnsNull()
+        {
+            var (outer, actual) = GetOuterOperationAndCorrespondingInnerOperation<SwitchStatementSyntax, ISwitchOperation, BreakStatementSyntax>(@"
+class C
+{
+    void F()
+    {
+        /*<bind>*/switch (1)
+        {
+            case 1:
+                for (;;)
+                {
+                    /*<bind>*/break;/*</bind>*/
+                }
+            break;
+        }/*</bind>*/
+        
+    }
+}", branch => branch.GetCorrespondingSwitch());
+
+            Assert.NotNull(outer);
+            Assert.Null(actual);
+        }
+
+        private void AssertOuterIsCorrespondingLoopOfInner<TOuterSyntax, TInnerSyntax>(string source)
+            where TOuterSyntax : SyntaxNode 
+            where TInnerSyntax : SyntaxNode
+        {
+            var (expected, actual) = GetOuterOperationAndCorrespondingInnerOperation<TOuterSyntax, ILoopOperation, TInnerSyntax>(
+                source, branch => branch.GetCorrespondingLoop());
+
+            Assert.Equal(expected.Syntax, actual.Syntax);
+        }
+
+        private void AssertOuterIsCorrespondingSwitchOfInner(string source)
+        {
+            var (expected, actual) = GetOuterOperationAndCorrespondingInnerOperation<SwitchStatementSyntax, ISwitchOperation, BreakStatementSyntax>(
+                source, branch => branch.GetCorrespondingSwitch());
+
+            Assert.Equal(expected.Syntax, actual.Syntax);
+        }
+
+        private (IOperation outer, IOperation corresponding) GetOuterOperationAndCorrespondingInnerOperation<TOuterSyntax, TOuterOp, TInnerSyntax>(
+            string source, Func<IBranchOperation, IOperation> findCorresponding)
+            where TOuterSyntax : SyntaxNode
+            where TOuterOp : class, IOperation
+            where TInnerSyntax : SyntaxNode
+        {
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.RegularWithFlowAnalysisFeature);
```

Comment:
>RegularWithFlowAnalysisFeature [](start = 82, length = 30)

You shouldn't need flow analysis for this.

Reply:
You should be fine with whatever the default `parseOptions` is, no need to pass an explicit parameter.

---
In reply to: [205600237](https://github.com/dotnet/roslyn/pull/28285#discussion_r205600237) [](ancestors = 205600237)